wgpu-hal/src/gles/egl.rs

```diff
@@ -225,6 +225,77 @@ enum SrgbFrameBufferKind {
     Khr,
 }
 
+pub struct AdapterContext {
+    glow_context: glow::Context,
+    egl: Arc<egl::DynamicInstance<egl::EGL1_4>>,
+    egl_display: egl::Display,
+    egl_context: egl::Context,
+    egl_pbuffer: Option<egl::Surface>,
+    mutex: Mutex<()>,
```

Comment:
why not `Mutex<glow::Context>`?

Reply:
Ah, that allows us to still lock the `glow::Context` even when calling `get_without_lock()` which will help a little more to protect against accidental concurrent access. Cool. :)